### DISCUSSION TITLE
Multiple users accessing Marimo

### DISCUSSION THREAD

**Discussion by jacksondelametter** [marimo-team/marimo#1664] [General]
I am working on a project where we want to deploy Marimo and allow multiple users be able to access it from a cloud environment. If I create a new Python environment for each user as well as start a Marimo instances using `marimo edit --port [port-number]` for each user to individually use, then each user can still see notebooks that other users are working on even though the Marimo instances are running in a separate Python environment and bound to a different port number. Is there a way to have multiple Marimo instances completely isolated from one another?

**Comment by akshayka** [comment]
@jacksondelametter , thanks for the question.

Are they all running on the same server/same container, or in independent containers? I think you could isolate the marimo edit servers entirely by running in different Docker containers. Would that be possible?

> each user can still see notebooks that other users are working on even though the Marimo instances are running in a separate Python environment and bound to a different port number

If they are running in the same container, I would guess it's it the recent files of other users that each of your users can see? Or what are they seeing?

**Comment by mscolnick** [comment]
@jacksondelametter you might find that you can achieve this with deploying marimo server's with JupyterHub

checking out this thread (near the bottom): https://github.com/marimo-team/marimo/issues/905
and this repo: https://github.com/jyio/jupyter-marimo-proxy

**Comment by dridk** [comment]
What about molab ?

**Comment by mikehogarth** [comment]
I would second the high value of molab as part of the Marimo offerings. I would go so far as to say I would pay to license molab even if Marimo itself is FOSS.
